### PR TITLE
Refactor get

### PR DESCRIPTION
**Pull request by peterchou3788**
Addresses: #7747 

Looked for areas where `_.get()` was used and see if object properties are suitable for refactoring. Based on `_.has()` issue, these changes can be updated to perform optional chaining and nullish coalescing operator.

Files with `.get()`:

- https://github.com/serverless/serverless/blob/053f5f420b45e9dec794e82d1bc23a2731a077ff/lib/plugins/aws/package/lib/saveServiceState.js
- https://github.com/serverless/serverless/blob/003696260c43acf2415fa6b05a212ea57bdec3d4/lib/plugins/aws/package/compile/events/apiGateway/lib/method/authorization.js
- https://github.com/serverless/serverless/blob/c9a6b75d49502cb429ad34ef4a917f6d286db60e/lib/plugins/aws/package/compile/events/apiGateway/index.js
- https://github.com/serverless/serverless/blob/f71649cdb5c141f2493b45d77c91d7e866d47fcf/lib/plugins/aws/deploy/lib/extendedValidate.js
-https://github.com/serverless/serverless/blob/9f0131fedf60e9104f38702d01e103b9a3b0f629/lib/plugins/aws/package/compile/events/websockets/lib/pickWebsocketsTemplatePart.js (Deep level properties so ignored)

### CODE REVIEW

File: lib/plugins/aws/deploy/lib/extendedValidate.js [hunk 1]
```diff
@@ -61,7 +61,6 @@ module.exports = {
       this.serverless.service.getAllFunctions().forEach(functionName => {
         const functionObject = this.serverless.service.getFunction(functionName);
         const individually =
-          _.get(functionObject, 'package.individually') ||
```
Comment (medikoo) [inline lib/plugins/aws/deploy/lib/extendedValidate.js:64]: Why it was removed?

File: lib/plugins/aws/deploy/lib/extendedValidate.js [hunk 2]
```diff
@@ -73,7 +72,7 @@ module.exports = {
           artifactFileName = this.provider.naming.getFunctionArtifactName(functionName);
           artifactFilePath = path.join(this.packagePath, artifactFileName);
 
-          if (_.get(functionObject, 'package.artifact')) {
```
Comment (medikoo) [inline lib/plugins/aws/deploy/lib/extendedValidate.js:76]: Are you sure it's safe? Is `package` guranteed to be set on `functionObject`?

File: lib/plugins/aws/package/compile/events/apiGateway/lib/method/authorization.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ const awsArnRegExs = require('../../../../../../utils/arnRegularExpressions');
 
 module.exports = {
   getMethodAuthorization(http) {
-    if (_.get(http, 'authorizer.type') === 'AWS_IAM') {
+      if (_.get(http, 'authorizer.type') === 'AWS_IAM') {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/method/authorization.js:8]: Are you sure it's safe (is `authorizer` guaranteed` to be set on `http` object?

File: lib/plugins/aws/package/lib/saveServiceState.js [hunk 4]
```diff
@@ -16,7 +16,6 @@ module.exports = {
     );
 
     const artifact = _.last(_.get(this.serverless.service, 'package.artifact', '').split(path.sep));
-
```
Comment (medikoo) [inline lib/plugins/aws/package/lib/saveServiceState.js:19]: Let's not introduce whitespace changes

File: lib/plugins/aws/deploy/lib/extendedValidate.js (hunk 2 above)
Comment (peterchou3788) [reply lib/plugins/aws/deploy/lib/extendedValidate.js:76]: Looking at the service class, the package is not guaranteed to be set. Therefore, there needs to be a check if package has a value other than null/undefined.

File: lib/plugins/aws/package/compile/events/apiGateway/lib/method/authorization.js (hunk 3 above)
Comment (peterchou3788) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/method/authorization.js:8]: In addition, looking at aws documentation, its optional for users to set an authorizer.

**Comment by peterchou3788** [comment]
Also is there an easier way to search for `.get`. I've been using the github advanced search to look for "get" as the keyword.

File: lib/plugins/aws/package/compile/events/apiGateway/lib/method/authorization.js (hunk 3 above)
Comment (medikoo) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/method/authorization.js:8]: AWS documentation is irrelevant here. We're processing Framework config

**Comment by medikoo** [comment]
> Most of the cases I found seem to not be safe for replacing. Does that mean the .get is necessary?

If they're not safe to replace, we should keep `_.get`

> Also is there an easier way to search for .get

Simply search code base in your editor for `_.get` references

**Comment by peterchou3788** [comment]
Hey, I have an issue with running `npm run lint`. When I run it, it tells me theres 20000+ errors due to single quote requirement for strings. I didn't encounter this error first time i ran it so I believe it must be a setting I changed or installed a lint version that altered the rules. I downloaded eslint as well using npm.

**Comment by medikoo** [comment]
> Hey, I have an issue with running npm run lint

Ensure that project is properly installed (simply by `npm install`) and everything should work as expected

**Comment by peterchou3788** [comment]
So I decided to make a new local copy of the repo and run the tests. This time it works as intended, but the prettier script always has a problem with a serverless.yml file that has a key `foo` and a value `foo` separated by a new line. Should I change this file as well or ignore it? 
Also, I have found more cases of `_.get` but they all seem to be required or unsafe. I don't need to changed the files that in node_modules because that's created when someone uses `npm install` correct?

**Comment by medikoo** [comment]
> This time it works as intended, but the prettier script always has a problem with a serverless.yml file that has a key foo and a value foo separated by a new line.

@peterchou3788 indeed, there was a small issue in prettier scripts configuration. It's fixed now in master.

Still you do not need to worry about files which you do not intend to update. In CI formatting is confirmed only on files to which you propose changes.

**Comment by medikoo** [comment]
@peterchou3788 great thanks for the effort! Still as there are no further updates, I'm going to close it. However if you still want to tackle it, I'll be happy to reopen

**Pull request closed without merging**